readthedocs/projects/views/private.py

```diff
@@ -347,7 +349,33 @@ def get_form_kwargs(self, step=None):
 
     def get_template_names(self):
         """Return template names based on step name."""
-        return "projects/import_{}.html".format(self.steps.current)
+        return f"projects/import_{self.steps.current}.html"
+
+    def process_step(self, form):
+        if isinstance(form, ProjectBasicsForm):
+            remote_repository = form.cleaned_data.get("remote_repository")
+            if remote_repository and remote_repository.vcs_provider == GITHUB:
+                for yaml in [
+                    ".readthedocs.yaml",
+                    ".readthedocs.yml",
+                    "readthedocs.yaml",
+                    "readthedocs.yml",
+                ]:
+                    try:
+                        response = requests.head(
+                            f"https://api.github.com/repos/{remote_repository.full_name}/contents/{yaml}",
+                            timeout=1,
+                        )
+                        if response.ok:
+                            log.info(
+                                "Read the Docs YAML file found for this repository.",
+                                yaml=yaml,
+                            )
```

Comment:
Should we also add a message to the user here? Something like `We detected a Read the Docs YAML file in your repository, triggering an initial build` or similar?

Reply:
Sounds good. Do you think that a `messages.add_message` would be enough here?

Reply:
Done in da2e9f5a7

![Screenshot_2024-08-19_17-30-37](https://github.com/user-attachments/assets/38a465ca-aab8-4819-b915-7e9d9ae7a890)